File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/ModelCompare.java
package org.kevoree.modeling.api;

import org.kevoree.modeling.api.trace.TraceSequence;

/**
 * Created by duke on 9/30/14.
 */

public interface ModelCompare {

    public void diff(KObject origin, KObject target, Callback<TraceSequence> callback);

    public void merge(KObject origin, KObject target, Callback<TraceSequence> callback);

    public void inter(KObject origin, KObject target, Callback<TraceSequence> callback);

}

File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/ModelCompare.kt
package org.kevoree.modeling.api

import org.kevoree.modeling.api.trace.TraceSequence

/**
 * Created by duke on 9/30/14.
 */

public trait ModelCompare {

    public fun diff(origin: KObject<*>, target: KObject<*>, callback: Callback<TraceSequence>)

    public fun merge(origin: KObject<*>, target: KObject<*>, callback: Callback<TraceSequence>)

    public fun inter(origin: KObject<*>, target: KObject<*>, callback: Callback<TraceSequence>)

}